py4DSTEM/process/diffraction/crystal.py

```diff
@@ -801,6 +808,31 @@ def calibrate_pixel_size(
             k_step
         )
 
+        # get discrete plot from structure factor amplitudes
+        int_exp = np.zeros_like(k)
+        k_px = (qr - k_min) / k_step
+        kf = np.floor(k_px).astype("int")
+        dk = k_px - kf
+
+        sub = np.logical_and(kf >= 0, kf < k_num)
+        int_exp = np.bincount(
+            np.floor(k_px[sub]).astype("int"),
+            weights=(1 - dk[sub]) * int_meas[sub],
+            minlength=k_num,
+        )
+        sub = np.logical_and(k_px >= -1, k_px < k_num - 1)
+        int_exp += np.bincount(
+            np.floor(k_px[sub] + 1).astype("int"),
+            weights=dk[sub] * int_meas[sub],
+            minlength=k_num,
+        )
+        int_exp = (int_exp ** bragg_intensity_power) * (k ** bragg_k_power)
+        sub = np.logical_and(
+            k >= k_min,
+            k <= k_max, 
+            )
+        int_exp /= np.max(int_exp[sub])
+
```

Comment:
This seems to replicate the calculate_bragg_peak_histogram function called immediately above, is there something additional that you would need from this section of the code?